tree/bst/bst.py

```diff
@@ -0,0 +1,189 @@
+"""
+Implement Binary Search Tree. It has method:
+    1. Insert
+    2. Search
+    3. Size
+    4. Traversal (Preorder, Inorder, Postorder)
+    5. Remove_max
+    6. Remove
+"""
+
+import unittest
+
+class Node(object):
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+class bst(object):
+    def __init__(self):
+        self.root = None
+
+    def get_root(self):
+        return self.root
+
+    """
+        Get the number of elements
+        Using recursion. Complexity O(logN)
+    """
+    def size(self):
+        return self.recur_size(self.root)
+
+    def recur_size(self, root):
+        if root is None:
+            return 0
+        elif root.left is None:
+            return 1 + self.recur_size(root.right)
+        elif root.right is None:
+            return 1 + self.recur_size(root.left)
+        else:
+            return 1 + self.recur_size(root.left) + self.recur_size(root.right)
+
+    """
+        Search data in bst
+        Using recursion. Complexity O(logN)
+    """
+    def search(self, data):
+        if self.root:
+            return self.recur_search(self.root, data)
+        else:
+            return False
+
+    def recur_search(self, root, data):
+        if root.data == data:
+            return True
+        elif root.left is None and root.right is None:
+            return False
+        elif data > root.data:     # Go to right root
+            return self.recur_search(root.right, data)
+        else:                      # Go to left root
+            return self.recur_search(root.left, data)
+
+    """
+        Insert data in bst
+        Using recursion. Complexity O(logN)
+    """
+    def insert(self, data):
+        if self.root:
+            return self.recur_insert(self.root, data)
+        else:
+            self.root = Node(data)
+            return True
+
+    def recur_insert(self, root, data):
+        if root.data == data:      # The data is already there
+            return False
+        elif data < root.data:     # Go to left root
+            if root.left:          # If left root is a node
+                return self.recur_insert(root.left, data)
+            else:                  # left root is a None
+                root.left = Node(data)
+                return True
+        else:                      # Go to right root
+            if root.right:         # If right root is a node
+                return self.recur_insert(root.right, data)
+            else:
+                root.right = Node(data)
+                return True
+
+    def remove(self, data):
+        if self.root:
+            return self.root.remove(data)
```

Comment:
There is no method as `remove` in class `Node`. Did you run your tests before committing?